app/src/pages/TokenPage/TokenPage.tsx

```diff
@@ -1,17 +1,85 @@
+  }, [updateCollectionToken]);
+
+  const updateCollection = React.useCallback(async (): Promise<void> => {
+    setCollection(undefined);
+    notdClient.retrieveCollection(registryAddress).then((retrievedCollection: Collection): void => {
+      setCollection(retrievedCollection);
+    }).catch((error: unknown): void => {
+      console.error(error);
+      setCollection(null);
+    });
+  }, [notdClient, registryAddress]);
+
+  React.useEffect((): void => {
+    updateCollection();
+  }, [updateCollection]);
 
   return (
-    <Stack direction={Direction.Vertical} isFullWidth={true} isFullHeight={true} childAlignment={Alignment.Center} contentAlignment={Alignment.Start} isScrollableVertically={true}>
-      <Text variant='header3'>{`Token Page for ${registryAddress} ${tokenId}`}</Text>
+    <Stack direction={Direction.Vertical} isFullWidth={true} isFullHeight={true} childAlignment={Alignment.Start} contentAlignment={Alignment.Start} isScrollableVertically={true}>
+      {collectionToken === undefined ? (
+        <LoadingSpinner />
+      ) : collectionToken === null ? (
+        <Text variant='error'>Collection Token failed to load</Text>
+      ) : (
+        <Stack direction={Direction.Horizontal} isFullWidth={true} childAlignment={Alignment.Start} contentAlignment={Alignment.Start} shouldAddGutters={true}defaultGutter={PaddingSize.Wide2} padding={PaddingSize.Wide2}>
+          <Box height='20rem' width='20rem' shouldClipContent={true}>
+            <Image source={collectionToken.imageUrl || defaultImage} alternativeText='image' fitType='contain' />
+          </Box>
+          <Stack direction={Direction.Vertical} isFullHeight={true} childAlignment={Alignment.Start} contentAlignment={Alignment.Center}>
+            <Text variant='header1'>{collectionToken.name}</Text>
+            <Stack direction={Direction.Horizontal} childAlignment={Alignment.Center} contentAlignment={Alignment.Center} shouldAddGutters={true}>
+              <Text>Owned By</Text>
+              <Box variant='rounded-borderColored' shouldClipContent={true} height='20px' width='20px'>
+                <Image source={'https://web3-images-api.kibalabs.com/v1/accounts/0x48e41913F2099300900cfcbB139F121429D38F5d/image' || defaultImage} alternativeText='Avatar' />
+                {/* TODO(Ajad-Abiola): implement Owned By: */}
+              </Box>
+              <Text>{truncateMiddle('0x48e41913F2099300900cfcbB139F121429D38F5d', 10)}</Text>
+            </Stack>
+            <Text>{`Last Bought for Ξ${TOKEN_TRANSFER.value / 1000000000000000000.0} at ${dateToString(TOKEN_TRANSFER.blockDate, 'HH:mm:yyyy')}`}</Text>
+            <Spacing variant={PaddingSize.Wide} />
+            <Stack direction={Direction.Horizontal} childAlignment={Alignment.Center} contentAlignment={Alignment.Center} shouldAddGutters={true}>
+              <Text>Part of</Text>
+              <Box variant='rounded-borderColored' shouldClipContent={true} height='20px' width='20px'>
```

Comment:
please make a component for this collection view - we can reuse it later (like the accountView we have). it should accept a single prop: `colletion: Collection`